### PR TITLE
Feat/vidispine item import

### PR DESCRIPTION
**Pull request by fredrikved**
From an XML file path, extract the file paths that should be imported to vidispine.

### CODE REVIEW

**Comment by fredrikved** [comment]
I haven't tested this yet.

File: services/transcode/merge.go [hunk 1]
```diff
@@ -172,7 +173,8 @@ func MergeSubtitles(input common.MergeInput, progressCallback ffmpeg.ProgressCal
 	// for each file, extract the specified range and save the result to a file.
 	for index, item := range input.Items {
 		file := filepath.Join(input.WorkDir, fmt.Sprintf("%d.srt", index))
-		cmd := exec.Command("ffmpeg", "-i", item.Path, "-ss", fmt.Sprintf("%f", item.Start), "-to", fmt.Sprintf("%f", item.End), "-y", file)
+		path := utils.IsilonPathFix(item.Path)
+		cmd := exec.Command("ffmpeg", "-i", path, "-ss", fmt.Sprintf("%f", item.Start), "-to", fmt.Sprintf("%f", item.End), "-y", file)
```
Comment (KillerX) [inline services/transcode/merge.go]:
```suggestion
		cmd := exec.Command(
		"ffmpeg", 
		"-i", path, 
		"-ss", fmt.Sprintf("%f", item.Start), 
		"-to", fmt.Sprintf("%f", item.End), 
		"-y", file,
		)
```

Makes it significantly easier to read the command IMO

**Review by KillerX — APPROVED** [review]

File: workflows/asset_ingest.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+		err = workflow.ExecuteActivity(ctx, vsactivity.ImportFileAsShapeActivity, vsactivity.ImportFileAsShapeParams{
+			AssetID:  vxID,
+			FilePath: file,
+			ShapeTag: "original",
+		}).Get(ctx, nil)
+		if err != nil {
+			return err
+		}
+	}
```
Comment (fredrikved) [inline workflows/asset_ingest.go:117]: This could be done in parallel, but it likely doesn't affect much or?

Comment (KillerX) [reply workflows/asset_ingest.go]: I don't think it really matters, no.

**Comment by fredrikved** [comment]
Flow has been tested now.

Only need to remove files when they are processed.

File: activities/vidispine/client.go [hunk 3]
```diff
@@ -13,3 +18,33 @@ func GetClient() *vidispine.VidispineService {
...
+		}
+		if job.Status != "STARTED" && job.Status != "READY" && job.Status != "WAITING" {
+			spew.Dump(job)
+			return fmt.Errorf("job failed with status: %s", job.Status)
+		}
+		activity.RecordHeartbeat(ctx, job)
+		if ctx.Err() != nil {
+			return ctx.Err()
+		}
+		time.Sleep(time.Second * 30)
```
Comment (KillerX) [inline activities/vidispine/client.go:48]: Is this correct or should the sleep be in temporal, so the job is not running while it is sleeping? Maybe we also need a "basic worker" that can do this kinds of non-process intensive stuff in parallel?

File: activities/common.go [hunk 4]
```diff
@@ -28,6 +28,9 @@ func newHeartBeater[T any](ctx context.Context) (chan struct{}, func(T)) {
 			select {
 			case <-timer.C:
 				activity.RecordHeartbeat(ctx, info)
+				if ctx.Err() != nil {
+					return
```
[the remark this answers is not in the document]
Comment (fredrikved) [reply activities/common.go:32]: The error is logged in temporal. This is mostly just to allow the activity to die if cancelled.

File: activities/vidispine/client.go [hunk 5]
```diff
@@ -13,3 +18,33 @@ func GetClient() *vidispine.VidispineService {
...
+		}
+		if job.Status == "FINISHED" {
+			return nil
+		}
+		if job.Status != "STARTED" && job.Status != "READY" && job.Status != "WAITING" {
+			spew.Dump(job)
+			return fmt.Errorf("job failed with status: %s", job.Status)
+		}
+		activity.RecordHeartbeat(ctx, job)
+		if ctx.Err() != nil {
```
[the remark this answers is not in the document]
Comment (fredrikved) [reply activities/vidispine/client.go:45]: Yes, exactly

File: activities/vidispine/client.go (hunk 3 above)
Comment (fredrikved) [reply activities/vidispine/client.go:48]: Most of what the worker does is basic, but yes temporal sleep is probably better.

**Pull request merged**